internal/backends/sqlite/metadata/registry.go

```diff
@@ -139,13 +161,14 @@ func (r *Registry) CollectionList(ctx context.Context, dbName string) ([]string,
 //
 // Returned boolean value indicates whether the collection was created.
 // If collection already exists, (false, nil) is returned.
+// If collection name doesn't pass backend level validation it returns ErrInvalidCollectionName.
```

Comment:
But it does not.

We also talked about how documentation for returned errors is fragile (because you have to update comments for all transitive callers) and how we will use contracts instead. Please don't document returned errors and error codes in comments.